hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/authority/CertificateStore.java

```diff
@@ -47,12 +50,21 @@ void storeValidCertificate(BigInteger serialID,
                              X509Certificate certificate) throws IOException;
 
   /**
-   * Moves a certificate in a transactional manner from valid certificate to
+   * Adds the certificates to be revoked to a new CRL and moves all the
+   * certificates in a transactional manner from valid certificate to
    * revoked certificate state.
-   * @param serialID - Serial ID of the certificate.
+   * @param certificates - List of X509 Certificates to be revoked.
+   * @param caCertificateHolder - X509 Certificate Holder of the CA.
+   * @param reason - CRLReason for revocation.
+   * @param securityConfig - Security Configuration.
+   * @param keyPair - Public and Private key of the CA.
    * @throws IOException
    */
-  void revokeCertificate(BigInteger serialID) throws IOException;
+  void revokeCertificates(List<X509Certificate> certificates,
+                          X509CertificateHolder caCertificateHolder,
+                          int reason, SecurityConfig securityConfig,
+                          KeyPair keyPair)
```

Comment:
can we abstract the CRL handling into a separate class (like Approver for CSR) without bringing CA keypair into the certificate store interface and class?